### PR TITLE
fix(agent-api): preserve original task text in Conversation

### PR DESCRIPTION
**Pull request by john-the-dev**
## What changed

`/tasks/active` now preserves the original `task:` text when rebuilding completed task rows from result files after the task file has already been archived.

Before this fix, result-only rows used the first line of the result body as `text`, so the Conversation Tasks list showed entries like `Done - ...` instead of the user's original request. The result body still remains available separately in `result`.

## Review first

- `src/agent-api.py`: `_task_display_fields_for_id()` and `_remember_done_result_file()`
- `tests/agent-api-task-display-text.test.py`: archived-task + result-only regression coverage

## User behavior / bug proven

Live repro before the patch on the dev instance showed `/tasks/active` returning completed rows like:

- `task-1783535563955.text = "Done ... addressed the review feedback on PR #2027."`
- `task-1783535277735.text = "Done ... screenshot is now attached to PR #2008."`

After applying the patch to the live dev backend and restarting only `agent-api`, the same endpoint returns the original requests:

- `task-1783535563955.text = "address comments: https://github.com/sonichi/sutando/pull/2027"`
- `task-1783535277735.text = "the screenshot should be attached to the pr: https://github.com/sonichi/sutando/pull/2008"`

## Tests run

- `python3 tests/agent-api-task-display-text.test.py` — passed
- `python3 tests/agent-api-task-field-injection.test.py` — passed
- `python3 tests/agent-api-pending-questions.test.py` — passed
- `python3 -m py_compile src/agent-api.py` — passed

## Edge cases checked

- Archived task file exists: completed row uses archived `task:` text and preserves `source`.
- No task file exists: completed row still falls back to the result summary as before.
- `POST /task-done` unknown-history path now also tries to recover the archived task text.

## Risk / rollback

Low risk. This changes only task-row display metadata for the web/client API; result contents and task completion semantics are unchanged. Rollback is reverting this commit.

### CODE REVIEW

**Comment by qingyun-wu** [comment]
### Live trajectory (independent reproduction)

Since the original dev-instance before-state is gone, here's a **deterministic black-box repro** that runs against the real `GET /tasks/active` HTTP endpoint on both the base and this PR. It simulates the exact scenario: the task file has already been archived, leaving only a result file whose first line is a `Done - …` summary.

**Fixture**
- `tasks/archive/task-999.txt` → `task: the screenshot should be attached to the pr: https://github.com/sonichi/sutando/pull/2008` (`source: discord`)
- `results/task-999.txt` → `Done - screenshot is now attached to PR #2008.`

**BEFORE (base `main`, unpatched) — `GET /tasks/active`**
```json
{
  "id": "task-999",
  "status": "done",
  "text": "Done - screenshot is now attached to PR #2008.",
  "result": "Done - screenshot is now attached to PR #2008.\n\nDetails."
}
```
→ `text` is the result summary (the bug).

**AFTER (this PR) — `GET /tasks/active`**
```json
{
  "id": "task-999",
  "status": "done",
  "text": "the screenshot should be attached to the pr: https://github.com/sonichi/sutando/pull/2008",
  "result": "Done - screenshot is now attached to PR #2008.\n\nDetails.",
  "source": "discord"
}
```
→ `text` is restored to the original request, and `source` is now preserved.

**Regression test** — `python3 tests/agent-api-task-display-text.test.py` passes on this branch (`agent-api task display text tests passed.`).

Repro is version-agnostic (drives the live endpoint, not the new internal helpers), so it exercises the actual user-facing path on both sides. Confirmed on `pr-2034` @ `3cf7708`.

**Comment by qingyun-wu** [comment]
Automated review finding:

`src/agent-api.py:137-144` only looks in `find_archived_task()` before backfilling the Conversation row title. But the normal `/task-done` callers post before the task file is archived: see `src/task-bridge.ts:799-809`, `823-833`, and `926-935`, where the archive happens 5s later. In that common path, `task-http-2` is still live under `tasks/`, `_task_display_fields_for_id()` returns `("", "")`, and `/task-done` still falls back to `result[:80]` at `src/agent-api.py:863-869`. So the UI will keep showing the result summary instead of the original `task:` text for the immediate status flip this PR is trying to fix. I think `_task_display_fields_for_id()` needs to probe the live task file first (or otherwise share the same lookup path as the task scanner), not just the archive.

File: src/agent-api.py [hunk 1]
```diff
@@ -120,6 +120,52 @@ def validate_twilio_signature(handler, body: str) -> bool:
...
+
+    if task_id not in task_history:
+        task_history[task_id] = {
+            "status": "done",
+            "text": display_text,
+            "time": result_file.stat().st_mtime,
+            "result": result_content,
+            "source": source_line,
+        }
+    elif task_history[task_id].get("status") != "done":
```
Comment (qingyun-wu) [inline src/agent-api.py]: Automated review: this still misses the race the PR is trying to fix. If `/tasks/active` sees `results/task-*.txt` before the corresponding task has been archived, the first call creates a done entry with the fallback result-summary text. Once the archive file appears, later scans stay on the `status == "done"` path and never repair `text`/`source`, so the Conversation row remains wrong until restart. I reproduced that sequence locally against this patch. Please keep backfilling when archived task metadata becomes available even for already-done entries.

**Comment by qingyun-wu** [comment]
Automated mergeability review (2026-07-09)

Signal: NOT MERGEABLE

Current blocker(s):
- Branch is behind current `main`; update/merge main and rerun checks before merging.

This is a conservative backlog triage signal. After the blockers are cleared, please rerun checks and request a focused review.

**Review by qingyun-wu — COMMENTED** [review]
Delta review of commit 84f9941 ("fix(agent-api): repair fallback done-row once task text becomes readable") since my 2026-07-08 23:25 review.

**Prior point — addressed**

The race I flagged (done row created from a result file before the task was archived keeps the fallback result-summary text until restart) is fixed: `_remember_done_result_file` (`src/agent-api.py:161-174`) now backfills `text` whenever the archived `task:` line becomes readable and differs from the cached value, and backfills `source` when missing — even for rows already in `status == "done"`. The new regression test (`tests/agent-api-task-display-text.test.py:130-163`) reproduces the exact poll-1-fallback / poll-2-repair sequence, including the archive appearing between polls. This matches the repro I ran against the previous revision.

Note the semantics change slightly beyond pure backfill: `text` is now force-synced to the archived `task:` line on every poll for any done row (not just fallback rows). Since both write paths derive `text` from the same `task:` parse, this converges rather than clobbers — fine as implemented, just worth knowing the archive is now authoritative for display text.

CI: all checks pass (tsc + tests, python diff coverage, guard jobs, CLA).

LGTM as a comment — no new issues in the delta.

**Review by liususan091219 — DISMISSED** [review]
Codex full-review sweep: checked against the 7 triage CR classes (source-grep tests, incomplete fix, band-aid, naming, false claim, live-test policy, hardcoded paths) - none fire. Approving.

`review-reason: codex-clean per full-review sweep`

**Comment by qingyun-wu** [comment]
LGTM. I diffed this against `main` and ran `python3 tests/agent-api-task-display-text.test.py` in the PR worktree; it passed, including the `/tasks/active` and `/task-done` HTTP-path regression coverage. The new `_remember_done_result_file()` reconciliation fixes both the archived-task backfill and the race where a fallback "done" title needed to be repaired on a later poll.

_Qingyun's Personal Codex Agent_

**Comment by liususan091219** [comment]
Before/after verification (origin/main 6388faf1 vs this head):

- **BEFORE (main)**: manual repro — completed tasks whose task file is archived show `text` = first line of the RESULT body in /tasks/active (`'task-done result body A'`, `'Result first line B'`) instead of the original ask. (Running the PR's full suite against main ERRORs on the PR-introduced helper — API drift, so the behavioral repro above is the bug proof.) HTTP-only subset of the test: AssertionError on main.
- **AFTER (this head)**: full suite PASS exit=0; same repro shows the ORIGINAL task text on both rows.

Verdict: fix is real. Also checked the #2103 overlap concern: no conflict — #2103 touched /status + /answer pending-questions parsing, this PR touches /tasks/active + /task-done result-row rebuild (disjoint regions). BEHIND main but MERGEABLE.

— Lucy (Mac Studio bot)
